components/portal/widgets/controllers.js

```diff
@@ -414,41 +414,43 @@ define(['angular'], function(angular) {
 
       var now = new Date();
       var templateIndex = indexWithinTemplateRange(now, callsToAction);
-      if (-1 < templateIndex) {
+      if (templateIndex > -1) {
         displayTimeSensitiveContent(callsToAction[templateIndex]);
       }
     };
 
     var resolveDate = function(str, defaultEndOfDay) {
-      var result = null;
-      if (str) {
-        var dateStr =
-          $filter('filterForDateWithYear')(str);
-        if (dateStr) {
-          if (dateStr.indexOf('T') < 0) {
-            if (defaultEndOfDay) {
-              dateStr += 'T23:59:59';
-            } else {
-              dateStr += 'T00:00:00';
-            }
-          }
-          result = new Date(dateStr);
+      // if it is valid ISO 8601 pass through
+      var fullISODate = moment(str, moment.ISO_8601, true);
+      if (fullISODate.isValid()) {
+        return fullISODate.toDate();
+      }
+
+      // if it is a partial ISO string, fill in missing fields
+      var partialISODate = moment(str, ['YYYY-MM-DD', 'MM-DD'], true);
+      if (partialISODate.isValid()) {
+        if (defaultEndOfDay) {
+          return partialISODate.endOf('day').toDate();
+        } else {
+          return partialISODate.startOf('day').toDate();
         }
       }
-      return result;
+
+      // this is not a valid date
+      return moment.invalid().toDate();
     };
 
     var indexWithinTemplateRange = function(now, callsToAction) {
       var result = -1;
+      // TODO: refactor as Array.prototype.findIndex when IE support is dropped
       // Check if today falls within any of the provided date ranges
       angular.forEach(callsToAction, function(callToAction, index) {
         if (result === -1) { // We haven't found a call to action
           var startDate =
             resolveDate(callToAction.activeDateRange.templateLiveDate);
           var endDate =
             resolveDate(callToAction.activeDateRange.templateRetireDate, true);
-          if (startDate && startDate < now // Template started
-            && (!endDate || endDate > now)) { // Template has not ended
+          if (moment().isBetween(startDate, endDate)) {
```

Comment:
This changed the behavior a little bit. If an endDate isn't specified, this comes up false. That's fine, that's how the previous code worked, but if we keep it like this, should change the documentation back

Reply:
Is this related to an outdated commit?

Reply:
nope. this line works like the pre- #592 code, in that it requires a `templateRetireDate` if there's a `templateLiveDate`. #592 changed that such that `templateRetireDate` was optional even if a `templateLiveDate`, and edited the documentation to say that.  Line 452 on the left hand side is checking whether the template has been retired, rather than whether now is between two valid dates.

This comment was just a poorly worded reminder to edit the documentation when changes in functionality occur.